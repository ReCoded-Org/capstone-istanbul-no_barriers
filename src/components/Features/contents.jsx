import laptopgirl from "./img1.jpg";
import emailgirl from "./img2.jpg";
import puzzle from "./img3.jpg";

const contents = [
  {
    title: "homePage.features.firstFeatTitle",
    src: laptopgirl,
    alt: "girl with Laptop",
    styling: {
      width: "45rem",
      hight: "34rem",
      top: "139.375rem",
    },
  },
  {
    title: "homePage.features.seondFeatTitle",
    src: emailgirl,
    alt: "girl with email and speaker",
    styling: {
      width: "45rem",
      hight: "37.5rem",
      left: "45rem",
    },
  },
  {
    title: "homePage.features.therdFeatTitle",
    src: puzzle,
    alt: "people fixing puzzle",
    styling: {
      width: "45rem",
      hight: "37.5rem",
    },
  },
];

export default contents;
