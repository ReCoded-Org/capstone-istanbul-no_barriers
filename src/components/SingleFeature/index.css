.featuresImg {
  max-width: 100%;
  height: 100%;
}

.featText {
  text-align: center;
}
