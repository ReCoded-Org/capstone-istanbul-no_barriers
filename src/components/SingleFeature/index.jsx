import React from "react";
import { Row, Col } from "react-bootstrap";
import "./index.css";

const SingleFeature = (props) => {
  return (
    <Row className="featContainer justify-content-center align-items-center">
      <Col lg={{ order: props.order }}>
        <img
          className="featuresImg img-fluid"
          src={props.src}
          style={props.styling}
          alt={props.alt}
        />
      </Col>
      <Col className="featText" lg="6">
        <h3>{props.title}</h3>
        <p>{props.text}</p>
      </Col>
    </Row>
  );
};

export default SingleFeature;
