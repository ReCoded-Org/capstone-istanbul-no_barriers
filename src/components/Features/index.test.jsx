import React from "react";
import renderer from "react-test-renderer";
import SingleFeature from "../SingleFeature/index";
import contents from "./contents";

it("Features renders correctly", () => {
  const tree = renderer
    .create(
      contents.map((content, index) => {
        if (index % 2 === 0) {
          return (
            <SingleFeature
              src={content.src}
              title={content.title}
              text={content.text}
              styling={content.styling}
              alt={content.alt}
              key={content.alt}
              order={"first"}
            />
          );
        } else {
          return (
            <SingleFeature
              src={content.src}
              title={content.title}
              text={content.text}
              styling={content.styling}
              alt={content.alt}
              key={content.alt}
              order={"last"}
            />
          );
        }
      })
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
