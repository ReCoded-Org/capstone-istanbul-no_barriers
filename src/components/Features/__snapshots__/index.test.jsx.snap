// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features renders correctly 1`] = `
Array [
  <div
    className="featContainer justify-content-center align-items-center row"
  >
    <div
      className="col-lg order-lg-first"
    >
      <img
        alt="girl with Laptop"
        className="featuresImg img-fluid"
        src="img1.jpg"
        style={
          Object {
            "hight": "34rem",
            "top": "139.375rem",
            "width": "45rem",
          }
        }
      />
    </div>
    <div
      className="featText col-lg-6"
    >
      <h3>
        homePage.features.firstFeatTitle
      </h3>
      <p />
    </div>
  </div>,
  <div
    className="featContainer justify-content-center align-items-center row"
  >
    <div
      className="col-lg order-lg-last"
    >
      <img
        alt="girl with email and speaker"
        className="featuresImg img-fluid"
        src="img2.jpg"
        style={
          Object {
            "hight": "37.5rem",
            "left": "45rem",
            "width": "45rem",
          }
        }
      />
    </div>
    <div
      className="featText col-lg-6"
    >
      <h3>
        homePage.features.seondFeatTitle
      </h3>
      <p />
    </div>
  </div>,
  <div
    className="featContainer justify-content-center align-items-center row"
  >
    <div
      className="col-lg order-lg-first"
    >
      <img
        alt="people fixing puzzle"
        className="featuresImg img-fluid"
        src="img3.jpg"
        style={
          Object {
            "hight": "37.5rem",
            "width": "45rem",
          }
        }
      />
    </div>
    <div
      className="featText col-lg-6"
    >
      <h3>
        homePage.features.therdFeatTitle
      </h3>
      <p />
    </div>
  </div>,
]
`;
